iModelCore/ECDb/ECDb/SchemaReader.cpp

```diff
@@ -287,20 +297,30 @@ ECClassP SchemaReader::GetClass(Context& ctx, ECClassId ecClassId) const
 
             case ECClassType::Relationship:
             {
-            ECRelationshipClassP newClass = nullptr;
-            if (schema.CreateRelationshipClass(newClass, className, false) != ECObjectsStatus::Success)
-                return nullptr;
-
             BeAssert(!stmt->IsColumnNull(relStrengthColIx) && !stmt->IsColumnNull(relStrengthDirColIx));
 
             Nullable<StrengthType> strengthType = SchemaPersistenceHelper::ToStrengthType(stmt->GetValueInt(relStrengthColIx));
             Nullable<ECRelatedInstanceDirection> strengthDir = SchemaPersistenceHelper::ToECRelatedInstanceDirection(stmt->GetValueInt(relStrengthDirColIx));
             if (strengthType.IsNull() || strengthDir.IsNull())
                 {
-                LOG.errorv("Failed to load ECRelationshipClass %s.%s. Its StrengthType or ECRelatedInstanceDirection is unsupported. The file might have been used with newer versions of the software.",
+                if (schema.OriginalECXmlVersionGreaterThan(ECVersion::Latest))
+                    {
+                    CacheSchemaElementWithUnknowns(TABLE_Class, ecClassId); // Schema is of a newer version and the relationship class contains new unknown features. Mark this class to not be loaded.
+                    m_cache.SetClassEntryToNull(ecClassId);
+                    LOG.warningv("The ECRelationshipClass %s.%s has a StrengthType or ECRelatedInstanceDirection which is unsupported. The file might have been used with newer versions of the software. The class will not be loaded.",
                            schema.GetName().c_str(), className);
+                    }
+                else
+                    {
+                    LOG.errorv("Failed to load ECRelationshipClass %s.%s. Its StrengthType or ECRelatedInstanceDirection is unsupported. The file might have been used with newer versions of the software.",

```

Comment:
I suggest changing the phrase "might have been used with newer versions", we actually know that it has been used with newer versions. I'd also avoid the term "file" since it could be read from memory. Something along the lines of "The schema uses a newer version of the standard than is known to the current software".

Reply:
Done!